components/authz-service/config/config_mgmt.go

```diff
@@ -0,0 +1,103 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/chef/automate/components/automate-deployment/pkg/toml"
+	base_config "github.com/chef/automate/lib/config"
+	event_ids "github.com/chef/automate/lib/event"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	RunningState    = "running"
+	NotRunningState = "not_running"
+)
+
+func defaultConfig() aggregateConfig {
+	return aggregateConfig{
+		ProjectUpdateStage: ProjectUpdateStage{
+			State: NotRunningState,
+			DomainServices: []ProjectUpdateDomainService{
+				ProjectUpdateDomainService{
+					Name:       event_ids.ComplianceInspecReportProducerID,
+					LastUpdate: time.Now(),
+				},
+				ProjectUpdateDomainService{
+					Name:       event_ids.InfraClientRunsProducerID,
+					LastUpdate: time.Now(),
+				},
+			},
+		},
+	}
+}
+
+// Manager - configuration manager for the service
+type Manager struct {
+	baseConfigManager *base_config.Manager
+}
+
+// Config - stores the configuration for the service
+type aggregateConfig struct {
+	ProjectUpdateStage ProjectUpdateStage `toml:"project_update_config"`
+}
+
+// NewManager - create a new config. There should only be one config for the service.
+func NewManager(configFile string) *Manager {
+	config := readinConfig(configFile, defaultConfig())
+	return &Manager{
+		baseConfigManager: base_config.NewManager(configFile, config),
+	}
+}
+
+// Close - to close out the channel for this object.
+// This should only be called when the service is being shutdown
+func (manager *Manager) Close() {
+	manager.baseConfigManager.Close()
+}
+
+// GetProjectUpdateStage - get the project update stage
+func (manager *Manager) GetProjectUpdateStage() ProjectUpdateStage {
+	return manager.baseConfigManager.Config.(aggregateConfig).ProjectUpdateStage
```

Comment:
I'm not totally familiar with this syntax-- is this a type check? Could it fail? At a quick glance it seems like a lot of calls chained